tests/Unit/Type/data/custom-query-type.php

```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+namespace My\Test\Extension;
```

Comment:
Should we use a more specific namespace here? I think this will break once other tests are added and use the same namespace.

Reply:
Good point, I just wanted to have something that is not part of the normal namespace and unique enough to work for now. But we can change that of course.

Reply:
I just realized that for the automatic namespace conversion to work, this has to end in "\Domain\Repository". I'll fix that

Reply:
I kept the test namespace for now, but added \Domain\Repository.